server/ai_process.go

```diff
@@ -334,6 +338,10 @@ func processAIRequest(ctx context.Context, params aiRequestParams, req interface
 		clog.Infof(ctx, "Error submitting request cap=%v modelID=%v try=%v orch=%v err=%v", cap, modelID, tries, sess.Transcoder(), err)
 
 		params.sessManager.Remove(ctx, sess)
+
+		if cctx.Err() == context.DeadlineExceeded {
```

Comment:
More idiomatic Go would be:

```
switch {
    case <-cctx.Done():
        break
    default:
}
```

This should check if the context done signal was fired (i.e. we hit `processingRetryTimeout`) otherwise we'll fall to the default block and continue into another loop iteration.

Reply:
Yondon. Thanks for your high quality feedback! Not sure why I implemented it like that 🤔. Your solution is a lot cleaner. Merged in https://github.com/livepeer/go-livepeer/commit/72dced7b710e407248f019117cee5e1777d00285e.